src/llscan.h

```diff
@@ -0,0 +1,111 @@
+#ifndef SRC_LLSCAN_H_
+#define SRC_LLSCAN_H_
+
+#include <lldb/API/LLDB.h>
+#include <map>
+#include <set>
+
+namespace llnode {
+
+class FindObjectsCmd : public CommandBase {
+ public:
+  ~FindObjectsCmd() override {}
+
+  bool DoExecute(lldb::SBDebugger d, char** cmd,
+                 lldb::SBCommandReturnObject& result) override;
+};
+
+class FindInstancesCmd : public CommandBase {
+ public:
+  ~FindInstancesCmd() override {}
+
+  bool DoExecute(lldb::SBDebugger d, char** cmd,
+                 lldb::SBCommandReturnObject& result) override;
+
+ private:
+  bool detailed_;
+};
+
+class MemoryVisitor {
+ public:
+  virtual ~MemoryVisitor(){};
+
+  virtual uint64_t Visit(uint64_t location, uint64_t available) { return 0; };
+};
+
+class TypeRecord {
+ public:
+  std::string* type_name;
```

Comment:
I think these should be private properties with public `inline` getters.

Another thing is a `*` here. It doesn't look like these two fields need to be allocated separately. What do you think about  removing `*`, and also initializing all of them from the constructor function?